course5/homework6/mystr.cc

```diff
@@ -0,0 +1,104 @@
+#include "mystr.h"
+#include <memory>
+#include <cmath> 
+#include <cstring>
+
+using namespace mystring;
+
+MyString::MyString()
+{
+	mStr = std::make_unique<char[]>(1); //length 1 for zero terminator
```

Comment:
This produces correct result, but it's not ideal. Ideally, the default constructor should not allocate (or do other expensive stuff).
Quite often you will find yourself creating default-constructed object, ad then, depending on runtime conditions, immediately assigning to it, or discarding it.

Consider this:
```
std::string text;
if (dataSource == DataSource::FILE)
   text = ReadFromFile(text);
else
   text = ReadFromMemory();
AnalyseText(text);
```
In the example above you see that the "content" of the default-constructed string is short lived.